src/reducers/reducer.js

```diff
@@ -0,0 +1,27 @@
+import { CITY_WEATHER } from "../actions";
+
+const dataReducer = (state = [], action) => {
+  switch(action.type) {
+    case CITY_WEATHER:
+      const city = action.payload.data.city;
+      const cityforecast = action.payload.data.list
+
+      return { 
+        ...state,
+        city: city.name,
+        id: city.id,
+        lat: city.coord.lat,
+        lon: city.coord.lon,
+        temp: cityforecast.map(function(t){
+          return t.main.temp}),
+        pressure: cityforecast.map(function(p){
+          return p.main.pressure}),
+        humidity: cityforecast.map(function(h){
+          return h.main.humidity})
+      }
```

Comment:
The data is getting wiped out on every search because you're turning this into an object every time. Instead, you'll want to use an array here and then add the new city object to it: `[...state, { city, id, lat, long, etc }]`